packages/server/modules/shared/repositories/roles.ts

```diff
@@ -0,0 +1,22 @@
+import { GetRoles, UpsertRole } from '@/modules/shared/domain/rolesAndScopes/operations'
+import { UserRole } from '@/modules/shared/domain/rolesAndScopes/types'
+import { Knex } from 'knex'
+
+let roles: UserRole[]
+
+export const getRoles =
+  ({ db }: { db: Knex }): GetRoles =>
+  async () => {
+    if (roles) return roles
+    roles = await db('user_roles').select('*')
```

Comment:
All of these col/table names should ideally have a DbSchema helper with properly typed column names, instead of magic strings

Reply:
The DbSchema helper has the knex instance tied into it As is I'd like to not use for the multi region repositories. 

As the DbSchema helper only gives us constants on the column names, it gives no type safety, it just gives us constants. 
With that said we'll always need integration tests for these functions to ensure correct behavior, as long as that is true, the internals i consider implementation detail. If a raw sql works it works. For now this works just fine without relying on an abstraction, that will possibly need to change.

Reply:
> As the DbSchema helper only gives us constants on the column names, it gives no type safety, it just gives us constants.

Not sure I understand you, cause it does give you type safety in the sense that you can't reference non-existent table/column names. Not using that also has all of the downsides of using string literal constants that we've already covered in the JS review guide.

It might need refactoring or decoupling from knex, but at the end of the day - using typed col/table names is def safer (and easier due to intellisense) than just dumping table/col names as magic strings everywhere